confgenerator/otel/processors.go

```diff
@@ -336,3 +336,31 @@ func ModifyInstrumentationScope(name string, version string) Component {
 		},
 	}
 }
+
+// GroupByGMPAttrs moves the "namespace", "cluster", and "location"
+// metric attributes to resource attributes. The
+// googlemanagedprometheus exporter will use these resource attributes
+// to populate metric labels.
+func GroupByGMPAttrs() Component {
+	return Component{
+		Type: "groupbyattrs",
+		Config: map[string]interface{}{
+			"keys": []string{"namespace", "cluster", "location"},
```

Comment:
Why aren't `instance_name` and `machine_type` included -- are they not resource attributes?

Reply:
Nope, they are just metric labels they are added for the prometheus receiver too so users can use those attributes in their queries. Full list of resource attributes are here: https://cloud.google.com/monitoring/api/resources#tag_prometheus_target